clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -652,7 +655,7 @@ public void subscribe(List<String> topics, ConsumerRebalanceListener listener) {
         try {
             log.debug("Subscribed to topic(s): {}", Utils.join(topics, ", "));
             this.subscriptions.subscribe(topics, listener);
-            metadata.setTopics(topics);
+            metadata.setTopics(subscriptions.groupSubscription());
```

Comment:
Why does consumers need to fetch the metadata for group subscription? If it is not subscribed to some topics their partitions will never to assigned to itself right?